src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,24 +1,136 @@
 package core.basesyntax;
 
-/**
- * <p>Реалізувати свою HashMap, а саме методи `put(K key, V value)`, `getValue()` та `getSize()`.
- * Дотриматися основних вимог щодо реалізації мапи (initial capacity, load factor, resize...)
- * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
- */
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+
+    private Node<K, V>[] table;
+    private float loadFactor;
+    private int size;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+        loadFactor = DEFAULT_LOAD_FACTOR;
+    }
 
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            putForNullKey(value);
+            return;
+        }
 
+        int index = indexFor(hash(key), table.length);
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (key.equals(node.key)) {
+                node.value = value;
+                return;
+            }
+            node = node.next;
+        }
+
+        addNode(key, value, index);
     }
 
     @Override
     public V getValue(K key) {
+        if (key == null) {
+            Node<K, V> node = table[0];
+            while (node != null) {
+                if (key == node.key) {
+                    return node.value;
+                }
+                node = node.next;
+            }
+        }
```

Comment:
can we remove logic inside if condition (key == null)? it seems to me, that it's already implemented in code below

Reply:
Yes we can after refactoring hash() method. 
Done.